From 4ad350e6a70b283577a290362c17e4eb4afc7022 Mon Sep 17 00:00:00 2001
From: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Date: Wed, 24 Oct 2012 18:05:22 +0200
Subject: [PORT FROM MAIN] wifi: prevent reading wlan chip register when FW recovery

BZ: 64299

Default behavior during recovery is to try to read the PC
register of the chip. On Intel architectures, this read is not
possible as SDIO bus is off at that stage. This ends up in a SDIO
timeout, slowing down the recovery process.
Fix this.

Change-Id: I7246d53e6faa8adb79f3e254d642ba2836930ccd
Orig-Change-Id: I864e2c8036d4e001f04738e6924f47e7b3d560c3
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Reviewed-on: http://android.intel.com:8080/71526
Reviewed-by: buildbot <buildbot@intel.com>
Reviewed-by: Casasnovas, QuentinX <quentinx.casasnovas@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Tested-by: lab_cactus <lab_cactus@intel.com>
Reviewed-on: http://android.intel.com:8080/78003
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wl12xx-compat/drivers/net/wireless/wl12xx/main.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
index 9196d9d..43cc17b 100644
--- a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
+++ b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
@@ -1426,11 +1426,13 @@ static void wl12xx_print_recovery(struct wl1271 *wl)
 	wl1271_info("Hardware recovery in progress. FW ver: %s",
 		    wl->chip.fw_ver_str);
 
-	ret = wl1271_read32(wl, SCR_PAD4, &pc);
-	if (ret < 0)
-		return;
+	if (wl->conf.fwlog.read_panic) {
+		ret = wl1271_read32(wl, SCR_PAD4, &pc);
+		if (ret < 0)
+			return;
 
-	wl1271_info("pc: 0x%x", pc);
+		wl1271_info("pc: 0x%x", pc);
+	}
 }
 
 static void wl1271_recovery_work(struct work_struct *work)
-- 
1.7.4.1

